compiler/src/iree/compiler/Codegen/Utils/Utils.cpp

```diff
@@ -711,29 +711,32 @@ OpFoldResult convertByteOffsetToElementOffset(RewriterBase &rewriter,
                                               Location loc,
                                               OpFoldResult byteOffset,
                                               Type elementType) {
-  OpFoldResult elementWidth =
-      TypeSwitch<Type, OpFoldResult>(elementType)
-          .Case<ComplexType, FloatType, IntegerType, VectorType>(
-              [&](auto type) -> OpFoldResult {
-                return rewriter.getIndexAttr(
-                    IREE::Util::getRoundedElementByteWidth(elementType));
-              })
-          .Default([&](Type t) -> OpFoldResult {
-            return rewriter.create<IREE::Util::SizeOfOp>(loc, elementType)
-                .getResult();
-          });
-  AffineExpr s0, s1;
-  bindSymbols(rewriter.getContext(), s0, s1);
-  return affine::makeComposedFoldedAffineApply(rewriter, loc, s0.floorDiv(s1),
-                                               {byteOffset, elementWidth});
+  if (isa<ComplexType, FloatType, IntegerType, VectorType>(elementType)) {
+    unsigned typeBitWidth = IREE::Util::getTypeBitWidth(elementType);
```

Comment:
this is wrong too - it doesn't support packing - you need to replicate what getRoundedPhysicalStorageSize is doing (I have a PR I was working on before randomized that does all the affine stuff - we'd want to use that here)

Reply:
I don't want users trying i3/i5/etc to hit compiler crashes because of this assert - I really think we should probably not add this verifier until we can do it properly)

Reply:
(got confused and thought this was lei's PR - but it has the same issue - I think this isn't the right expression and I'm not a fan of the compiler crashing if a user tries to use i3/i5)

Reply:
Well, we don't have a good representation of i3/i5 etc. We really can't support those right now. That's why the assert so that we can resist when we can